stake-pool/program/src/processor.rs

```diff
@@ -1945,6 +1944,25 @@ impl Processor {
                 stake_space,
             )?;
 
+            // if needed, pre-fund the rent-exempt reserve from the reserve stake
+            let required_lamports_for_rent_exemption =
+                stake_rent.saturating_sub(source_transient_stake_account_info.lamports());
+            if required_lamports_for_rent_exemption >= reserve_stake_info.lamports() {
+                return Err(StakePoolError::ReserveDepleted.into());
+            }
```

Comment:
this case is probably not worth caring about but note it fails if the reserve is empty but the transient account is fully funded

Reply:
I see what you mean... the logic is really touchy around here. Note that the reserve shouldn't ever be empty, since it needs to have at least the rent-exemption to exist, so this line shouldn't ever trip unnecessarily.

I put this in because some later tests showed that I was accidentally drawing everything out of the reserve and destroying it, so the intent is to make sure that we don't delete the reserve ever.

How about this then?

```
if required_lamports_for_rent_exemption > 0 {
    if required_lamports_for_rent_exemption >= reserve_stake_info.lamports() {
        return Err(StakePoolError::ReserveDepleted.into());
    }
    Self::stake_withdraw(.....)?; 
}
```